quickwit-search/src/root.rs

```diff
@@ -216,9 +219,8 @@ fn jobs_to_leaf_request(
         search_request: Some(request_with_offset_0),
         split_metadata: jobs
             .iter()
-            .map(|job| {
-                extract_split_and_footer_offsets(split_metadata_map.get(&job.split_id).unwrap())
-            })
+            .filter_map(|job| split_metadata_map.get(&job.split_id))
```

Comment:
Note: unwrap is unsafe. It's better to check with filter*

Reply:
I might surprise you but I think I prefer the unwrap here  :).
But you are right, maybe something else would be even better.

So by unsafe I think you meant it can panic. (unsafe usually means something else).
This is right. If one of the key in that map is missing, then we get a panic... which is better 
than silently returning search results, without spotting the bug.

So the role of that unwrap is exactly the same as an `assert`.
It is a bit lame though because it is impossible for the reader to know if it is a "valid" unwrap or a  "sloppy" unwrap.
(The assert is about an invariant entirely enforced in the root so it is justified.)

Maybe we could replace it by an expect with a message explaining the situation, 
or use filter_map and add an explicit assert on the length of split_metadata.

```
let split_metadata = jobs.iter().filter_map(|job| {} /*...*/
assert_eq!(split_metadata.len(), jobs.len());
```

Reply:
I love the idea of having `assert` here for 2 reasons:
- Resilient to user when no need for debugging
- Easier to troubleshoot when turning on debug-mode
Let me update the code